core/src/main/java/org/apache/brooklyn/entity/stock/OptionalEntityImpl.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.stock;
+
+import java.util.Collection;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.brooklyn.api.entity.EntitySpec;
+import org.apache.brooklyn.api.location.Location;
+
+public class OptionalEntityImpl extends BasicStartableImpl implements OptionalEntity {
+
+    @Override
+    public void start(Collection<? extends Location> locations) {
+        Boolean create = config().get(CREATE_OPTIONAL_ENTITY);
+        if (Boolean.TRUE.equals(create)) {
+            EntitySpec<?> spec = Preconditions.checkNotNull(config().get(OPTIONAL_ENTITY_SPEC));
+            addChild(EntitySpec.create(spec));
+        }
+        super.start(locations);
+    }
```

Comment:
Check if the entity is already created - to handle start -> stop -> start/restart.